### ISSUE TITLE
[PROPOSAL] Improving public engagement for OpenSearch release process

### ISSUE THREAD

**Issue by bbarani** [opensearch-project/.github#167]
<h3>Summary:</h3>OpenSearch is a fast-growing community-contributed project and currently has around 600+ contributors who actively participate in the growth of the project day-to-day. OpenSearch has been piloting the process of hosting public meetings for some time, where public members can join the meetings and actively participate synchronously. Being an open source project, it is imperative that we move towards an open governance model to establish shared ownership of OpenSearch with the public in the day-to-day project management tasks. This proposal focuses on getting feedback on proposed release process, roles, mechanism along with release process documentation to improve the public engagement for OpenSearch releases.<br><h3><b>Motivation:</b></h3><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">OpenSearch currently release major, minor and patch versions on a regular cadence as listed on this </span><a href="https://opensearch.org/releases.html">release page</a><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">. The infrastructure to build, test and release artifacts are available at </span><a href="https://build.ci.opensearch.org/">https://build.ci.opensearch.org/</a><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">. A public release GitHub issue using this </span><a href="https://github.com/opensearch-project/opensearch-build/blob/main/.github/ISSUE_TEMPLATE/release_template.md">release template</a><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328"> is created in the </span><b><a href="https://github.com/opensearch-project/opensearch-build">opensearch-build repo</a></b><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328"> along with individual component GitHub issues created using this </span><a href="https://github.com/opensearch-project/opensearch-build/blob/main/.github/ISSUE_TEMPLATE/component_release_template.md">template</a><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328"> for all plugins participating in the release.</span> <span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">A primary release manager along with secondary release managers corresponding to participating repos are assigned for a release. The primary release manager goes over the individual release issues periodically across the repos and engage the secondary release managers to take appropriate action as needed throughout the release process. A release call is scheduled with necessary stakeholders to complete the release process.</span><br><br>The status of a release is currently tracked in public GitHub issue, but we believe there is additional opportunity for public collaboration during releases. We strongly believe that involving anyone who wants to be involved in the public is an important step an important step to improve the trust and transparency within the public along with getting additional support, feedback from external contributors for closing the gaps, and improvements to the overall release process.<br><br>This scope of this RFC doesn’t include standalone OpenSearch client releases that has been already fully automated and self-serviceable by anyone as described in this <a href="https://github.com/opensearch-project/opensearch-build/issues/1234">GitHub issue</a>.<br><br>It would be helpful to go over the current roles and responsibilities before diving deep in to the release process.<br><h3>Roles and responsibilities:</h3>The below roles and responsibilities are currently involved in OpenSearch release cycle process, we want to collaborate with the public to improve it further by adding, modifying roles and responsibilities (as needed).<br>

Role | Responsibilities
-- | --
Release manager |  Responsible for planning, testing, tracking, release, deployment, communication, and risk management.<br> Collaborate with technical and project management committee to finalize a release schedule and plan for OpenSearch <br> Responsible for managing the release lifecycle process that includes, <br> * Scheduling the release * Co-ordinating between teams <br> * Broadcasting release candidate information to gather votes <br>  * Execute automated tests <br> * Signing the artifacts <br>  * Deploying release artifacts as per the schedule <br> * Completing post release activities <br> Maintain proper coordination between multiple participating teams to update the project related information in publicly accessible platforms
Repository owner | Responsible for assigning a repo level release manager / POC for a specific release<br>Work closely with release manager to identify, remediate possible gaps for a release corresponding to a specific feature on this repo<br>Ensure sanity tests are executed and documented by assigned release manager<br>Help remediate issues found during testing<br>Surface any gaps to release manager in timely manner<br>Provide votes for finalizing release candidate
Maintainers | Sanity test the release candidate Surface any issues, gaps to release manager in timely manner<br>Help remediate issues found during testing<br>Provide votes for finalizing release candidate
Project management committee | The Project Management Committee is responsible for high-level technical direction of the product<br> The roadmap of the OpenSearch project and releases<br>Creating appropriate Working Groups ( User Feedback, Documentation etc..) to gather the necessary public feedback before making decisions.Technical resources (e.g., code repositories, servers)<br>Maintaining maintainers, committers list<br>Call for closing vote / vote to table any issue<br>Go / No-Go vote on product releases

<h3>OpenSearch / OpenSearch dashboards Release:</h3>At a high level. OpenSearch release process involves collaboration with multiple maintainers from OpenSearch, OpenSearch dashboards, website, documentation GitHub repositories that are participating in a release. OpenSearch follows de-centralized development model where the development for core engine, plugins happens across different geographical location. The distribution build (<a href="https://build.ci.opensearch.org/job/distribution-build-opensearch/">OpenSearch</a>, <a href="https://build.ci.opensearch.org/job/distribution-build-opensearch-dashboards/">OpenSearch Dashboard</a>s) integrates the code across all these repositories for all supported versions via automated CI / CD process to generate, test and publish nightly builds. <br><br>These are the pre-requisites that needs to be closed out before the release cycle process.<br><h4>Pre-requisites:</h4><ol><li style="list-style-type:decimal">Release manager for a specific OpenSearch release is finalized and assigned 28 days before a planned release.</li><ol><li style="list-style-type:lower-alpha">Once the release manager is finalized, his details will be assigned to the release GitHub issue created in  <a href="https://github.com/opensearch-project/opensearch-build">opensearch-build</a> GitHub repository and also broadcasted to <a href="https://opensearch.slack.com/archives/C0561HRK961">public slack channel</a>.</li></ol><li style="list-style-type:decimal">Release issue corresponding to major, minor, patch is created using the<a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/.github/ISSUE_TEMPLATE/release_template.md"> <b>release template</b></a>.</li><ol><li style="list-style-type:lower-alpha">Release issue is immediately after the OpenSearch core bumps the version to next major, minor, patch version. </li></ol><li style="list-style-type:decimal">Release manager is responsible for creating <b><a href="https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#components-release-issue">child component issues</a></b> on all Github repositories participating in a release.</li><li style="list-style-type:decimal">Release manager is responsible for releasing both OpenSearch and OpenSearch dashboards.</li><ol><li style="list-style-type:lower-alpha">This is subject to change in future as OpenSearch and OpenSearch dashboards are technically independent products with their own release cycle process.</li></ol><li style="list-style-type:decimal">Release manager will obtain all required access during release cycle process.. </li><ol><li style="list-style-type:lower-alpha">This step needs to be completed 5 days before the scheduled preparation phase for a release. Release manager works with the members of <a href="https://github.com/opensearch-project/opensearch-build">opensearch-build</a> team to get all necessary access. They can also reach out to the members via OpenSearch <a href="https://opensearch.slack.com/archives/C0561HRK961">public slack channel</a> (#release)</li></ol><li style="list-style-type:decimal">Release manager will co-ordinate with all participating GitHub repo maintainers via OpenSearch <a href="https://opensearch.slack.com/archives/C0561HRK961">public slack channe</a> (#release)</li><li style="list-style-type:decimal">Any questions, issues, feedback related to the release cycle, process, mechanism is discussed in OpenSearch <a href="https://opensearch.slack.com/archives/C0561HRK961">public slack channe</a> (#release)</li></ol><h4>Proposed release cycle process:</h4><u><b>Note</b></u>: The <a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md#post-release">PR</a> that describes the end to end OpenSearch release process is in review. The information links added to each of the following steps will be part of the <a href="https://github.com/opensearch-project/opensearch-build">opensearch-build</a> GitHub repository once the PR is merged.<br><br><ul><li><b><a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md#preparation">Preparation </a>(14 days from release date) - Owner  [<i>Repository owner, PMC committee, </i>Release manager ]</b></li><ul><li>This phase involves preparing for a release that includes assigning a release owner, updating the release cycle dates, finalizing the feature for a release as listed on the roadmap along with creating component release issues across all participating repos in a release.</li></ul><li><b><a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md#release-branch-readiness">Release branch readiness</a> <i>(Not applicable for patch release)   (10 days from release date) - Owner  [ Repository owner, Release manager ]</i></b></li><ul><li>This phase requires that every team participating in a release have their release (candidate) branch created by this date. The distribution build workflow will also start using the release branch to create release candidate instead of <code>.x</code> branches.</li></ul><li><b><a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md#code-complete-and-feature-freeze">Code Complete / Feature freeze</a>   (7 days from release date)  [Release manager, Repository owner]</b></li><ul><li>All maintainers of OpenSearch GitHub repositories targeting a specific release merge their PR’s ( i.e.e vMajor.minor.patch labeled PRs) to release branch branch before release manager creates the release candidate.</li></ul><li><b><a href="https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#release-candidate">Release candidate creation</a><span textcolor="#24292f" style="--original-color: #24292f;color: #24292f">  (6 days from release date) [Release manager]</span></b></li><ul><li><span textcolor="#24292f" style="--original-color: #24292f;color: #24292f">The automated build workflows will create release candidates with the finalized commit and publish the docker images and associated distribution artifacts for testing. The release manager will broadcast the release candidate to public for sanity tests. </span></li></ul><li><b><a href="https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#release-candidate-creation-and-testing">Release testing</a>  (5 - 2 days from release date) [Release manager, Maintainers, Repository owners]</b></li><ul><li>Automated integration, BWC and performance testing is kicked off against the release candidate. The release candidate information is also broadcasted to wider group for another round of smoke testing using the standard communication template. </li></ul><li><b><a href="https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#go-or-no-go">Go / No-Go voting</a> </b><b>(1 day from release date) [Release manager, Project management committee, Repository owners]</b></li><ul><li>After all validations and verification, promote the artifacts to production bucket for the final release after getting the sign off from Project management<b> </b> committee through <b><i>Go / No-Go voting</i> </b>process. </li></ul><li><a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md#release"><b>Release</b></a><b>  (Release day!) [Release manager, Project management committee, Repository owners ]</b></li><ul><li>Release the artifacts to production distribution channels, work with website team to update the website and inform the public of the release. </li></ul><li><b><a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md#post-release">Post release activities</a> (Release day!) [Release manager]</b></li><ul><li>This steps ensures that the tags are created for all participating repos, incrementing versions for next patch, minor release along with house cleaning activities.  Release manager also collaborates with the website and documentation repo owners to publish the website updates, blog posts and documentation update corresponding to a release.</li></ul></ul><br><b><u>Note: </u></b>We have put out a <a href="https://github.com/opensearch-project/.github/issues/150">proposal</a> to <b>move away from release train model</b> with defined release date to <b>release window model with a defined release candidate (RC) creation date</b>. We will iterate on “Release candidate” creation along with “Release testing” phase until we finalize the release candidate. <b>Release day will be 2 days after then final RC creation date. </b>Project management<b> </b> committee can also recommend to skip a release version if we are not able to finalize the release candidate within a reasonable time period to avoid cascading effect to releases.<br><hr/><h3>Frequently asked questions (FAQ):</h3><h4>How are the release managers finalized for a release?</h4>Release managers will be finalized based on volunteer model. A request for release managers for a specific release will be broadcasted in GitHub as we well as OpenSearch Slack channel and the release manager will be selected based on the first come first serve basis. <br><br><u><b>Note</b></u>: Release managers has to be a maintainers of Github repositories under OpenSearch project.<br><h4>Where can I see the finalized features for a specific release ?</h4>You can visit OpenSearch public <a href="https://github.com/orgs/opensearch-project/projects/1">roadmap</a> to get the comprehensive list of features scheduled to be released in specific OpenSearch version.<br><h4>Does OpenSearch and OpenSearch dashboards have separate release timeline ?</h4>Currently OpenSearch and OpenSearch dashboards are released at same time (with same version number) as part of OpenSearch release. <br><h4>How can I participate in release process?</h4>If you are not part of the listed roles, you can still actively participate in release process by sanity testing the release candidate along with providing your feedback, votes for finalizing the release candidate.<br><h4>Who are the members in current project management committee?</h4><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">The current membership is seeded by a specific group of AWS employees actively contributing to OpenSearch project. </span><br><h4>How can I become a member of project management committee?</h4><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">Additional appointments to project management committee will happen in near future based on open governance model.  We would like to get your feedback on the process, mechanism to add additional members to project management committee. </span><br><h4>How can I become a maintainer of a repository?</h4><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">Any new members who are interested in becoming maintainers on specific Github repository need to be nominated by the current maintainers through voting process as described </span><a href="https://github.com/opensearch-project/.github/blob/main/RESPONSIBILITIES.md#becoming-a-maintainer">here</a><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">. </span><br><hr/><h4>Next steps:</h4>All the steps listed above in release cycle process are already accessible by the public and the release steps, process are documented in public Github issue. One of the primary focus of this RFC is to surface the release process, roles and responsibilities along with detailed release guide to improve public engagement in OpenSearch release process.<br><br>We would like your feedback on,<br><ul><li>Improving the public engagement for OpenSearch release process</li><ul><li>We would like a broader participation in OpenSearch release cycle process. We have updated the <a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/.github/ISSUE_TEMPLATE/release_template.md">release template</a>, <a href="https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md">SOP</a> and collaboration <a href="https://opensearch.slack.com/archives/C0561HRK961">channels</a> to make the OpenSearch release process more accessible by the public. We need your feedback on any gaps, process technical limitations that would help improve the engagement.</li></ul><li>Formulating process, mechanism to add new roles to release process</li><ul><li>Please provide us feedback to draft a proposal to add, modify roles to OpenSearch release process</li></ul><li>Improving, simplifying the steps involved in current release cycle process</li><ul><li>Help us improve the OpenSearch release process by providing your inputs. We welcome both technical and process inputs.</li></ul><li>Improving the quality of OpenSearch artifacts</li><ul><li>Provide us inputs on improving the quality bar of every OpenSearch releases.</li></ul><li>Any other feedback related to release process, roles and responsibilities</li><ul><li>Feel free to provide any other ad-hoc inputs related to release process, public engagement, roles and responsibilities.</li></ul></ul><span textcolor="#1f2328" style="--original-color: #1f2328;color: #1f2328">We would like get your feedback and suggestions on this RFC to make the release process as seamless as possible. We look forward to collaborating with the public.</span><br>

**Comment by acarbonetto** [comment]
Thanks @bbarani for this document!  Having everyone aligned on a release process will be extremely helpful. A couple of questions/requests: 

1. Could we create a separate release channel for each new release.  For example: #release_2.8.0, #release_2.9.0, #release_2.8.1, etc... so that we stay on-topic and don't have any confusion.  We can archive the channel once release retrospective is completed for all parties. 
2. Can we start the release process for dependency libraries earlier, while plugins and repos without dependencies start later?  `common-utils`, for example: we could get release ready earlier so that all the plugins (such as `ml-commons`) depending on `common-utils` are unblocked.  
3. Report breaking API and library changes to the release channel to solicit feedback and notify dependent plugin owners of the change(s). 
4. Advertise the changes and comments from release issues (see https://github.com/opensearch-project/opensearch-build/issues/3434) into the release channel for slack notifications.

**Comment by peternied** [comment]
Thanks for documenting all of this the clarity is a great improvement - could you transform this issue's description into a pull request on the `RELEASING.md` so this is codified and committed to the repository?

**Comment by nknize** [comment]
> Thanks for documenting all of this the clarity is a great improvement - could you transform this issue's description into a pull request on the `RELEASING.md` so this is codified and committed to the repository?

+1.. let's get this into an actual reviewable PR so we can comment inline. 

Can we add some concrete language to some of these steps so folks know what goes into each one?

For example: 
1. `Release manager for a specific OpenSearch release is finalized and assigned 28 days before a planned release.`  
    * What's the process here? Do we request volunteers on slack #releases channel?
2. Once the release manager is finalized, his details will be assigned to the release GitHub issue created in [opensearch-build](https://github.com/opensearch-project/opensearch-build) GitHub repository and also broadcasted to [public slack channel](https://opensearch.slack.com/archives/C0561HRK961). 
    * Who does this? The release manager? Assigns themselves as the assignee in github? Posts a slack message?
3. Release manager is responsible for creating [child component issues](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#components-release-issue) on all Github repositories participating in a release.
    * Can we automate the process if it isn't already? If it is automated, is there a doc the RM can use to show them how to do this?

Let's start with that here and then have one single FOOLPROOF page an RM can follow even if they've never come close to the release process.

**Comment by prudhvigodithi** [comment]
> 2\. Can we start the release process for dependency libraries earlier, while plugins and repos without dependencies start later?  `common-utils`, for example: we could get release ready earlier so that all the plugins (such as `ml-commons`) depending on `common-utils` are unblocked.

Hey @acarbonetto, it should be possible to onboard plugins earlier provided the version increment for [core](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#core-version-increment) and [common-utils](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#components-version-increment) is completed and added to the [input manifest](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#input-manifest), this depends on how quick/early an RM and the plugin team coordinate to achieve this. We can also think of an automation where the plugins that does not have dependencies are force incremented along with core (should be ok since we are releasing all the plugins along with core as a single bundle, rather than an isolated release for each plugin).

**Comment by prudhvigodithi** [comment]
> Release manager is responsible for creating [child component issues](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#components-release-issue) on all Github repositories participating in a release.
> * Can we automate the process if it isn't already? If it is automated, is there a doc the RM can use to show them how to do this?

Even though this is not required for patch release, +1 for the automation where the [component release issues](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#components-release-issue) are created along with the main [release issue](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#release-issue) part of the build repo.

**Comment by reta** [comment]
@bbarani thanks a lot for the proposal, I think this is a HUGE step in right direction, I would like to bring a few notes / questions:
 - at the moment, all related release builds are happening on Jenkins (https://build.ci.opensearch.org/), this is a publicly available instance (:clap: ) but I suspect release manager might need to rerun some builds on demand, either because of hiccups or new fixes, this is not possible at the moment for non-AWS collaborators
 - at the moment, the release tracking includes the components under OpenSearch project only (please correct me if I am wrong), but I suspect the community ( plugins, and in future, extensions authors) may also be interested in including their  project into the release (fe, to increase the community awareness), would that make sense?

**Comment by bbarani** [comment]
Thanks for your feedback @reta . Please find my comments inline.

> @bbarani thanks a lot for the proposal, I think this is a HUGE step in right direction, I would like to bring a few notes / questions:
> 
> * at the moment, all related release builds are happening on Jenkins (https://build.ci.opensearch.org/), this is a publicly available instance (👏 ) but I suspect release manager might need to rerun some builds on demand, either because of hiccups or new fixes, this is not possible at the moment for non-AWS collaborators

Currently only specific group of folks under AWS have access to execute, modify builds but we will be expanding the access to all (including external members) release managers as well.

> * at the moment, the release tracking includes the components under OpenSearch project only (please correct me if I am wrong), but I suspect the community ( plugins, and in future, extensions authors) may also be interested in including their  project into the release (fe, to increase the community awareness), would that make sense?

We currently do not include any plugins outside of OpenSearch organization in to distribution bundle. We don't have a process to include external plugins that lists the SLA, roles and responsibilities yet.  @dblock @elfisher might be able to provide more insights on future plans.

Having said that, the release [manifests](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.9.0/opensearch-2.9.0.yml) dictates what needs to be in a release bundle and the limitation on including external plugins is more related to the process limitation rather than technical limitation.

**Comment by bbarani** [comment]
> Thanks for documenting all of this the clarity is a great improvement - could you transform this issue's description into a pull request on the `RELEASING.md` so this is codified and committed to the repository?

Sure, we will add the summary along with roles and responsibilities to this [PR](https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md) soon. CC: @prudhvigodithi

**Comment by bbarani** [comment]
> > Thanks for documenting all of this the clarity is a great improvement - could you transform this issue's description into a pull request on the `RELEASING.md` so this is codified and committed to the repository?
> 
> +1.. let's get this into an actual reviewable PR so we can comment inline.
> 
> Can we add some concrete language to some of these steps so folks know what goes into each one?
> 
> For example:
> 
> 1. `Release manager for a specific OpenSearch release is finalized and assigned 28 days before a planned release.`
>    
>    * What's the process here? Do we request volunteers on slack #releases channel?

Yes, the plan is to request for release managers for upcoming release immediately after the current release as part of post release activities. We will update the post release activities section in [RELEASE_PROCESS_OPENSEARCH.md](https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md) with these details.

> 2. Once the release manager is finalized, his details will be assigned to the release GitHub issue created in [opensearch-build](https://github.com/opensearch-project/opensearch-build) GitHub repository and also broadcasted to [public slack channel](https://opensearch.slack.com/archives/C0561HRK961).
>    
>    * Who does this? The release manager? Assigns themselves as the assignee in github? Posts a slack message?

Release manager will not have access to the build repo to assign himself to the GitHub release issue but members of @opensearch-project/engineering-effectiveness will help with assigning them to release issue.

> 3. Release manager is responsible for creating [child component issues](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#components-release-issue) on all Github repositories participating in a release.
>    
>    * Can we automate the process if it isn't already? If it is automated, is there a doc the RM can use to show them how to do this?

@prudhvigodithi is looking in to automating this process so that the child issues are automatically created on all plugin repos when parent release issue is created.

> 
> Let's start with that here and then have one single FOOLPROOF page an RM can follow even if they've never come close to the release process.

**Comment by prudhvigodithi** [comment]
> Release manager might not have access to the OpenSearch build repo to self-assign the GitHub release issue but members of @opensearch-project/engineering-effectiveness will help with assigning them to release issue.

I have added a new section [Upcoming Release Preparation](https://github.com/opensearch-project/opensearch-build/blob/d7149d7a696495865053e7700951336dded2f21e/RELEASE_PROCESS_OPENSEARCH.md#upcoming-release-preparation) to handle this.

**Comment by prudhvigodithi** [comment]
> Sure, we will add the summary, motivation along with roles and responsibilities to this [RELEASE_PROCESS_OPENSEARCH.md](https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md) soon. CC: @prudhvigodithi

@bbarani should this be part of the https://github.com/opensearch-project/.github/blob/main/RELEASING.md file as the file `RELEASE_PROCESS_OPENSEARCH.md` can focus on just the release steps.

**Comment by bbarani** [comment]
> > Sure, we will add the summary, motivation along with roles and responsibilities to this [RELEASE_PROCESS_OPENSEARCH.md](https://github.com/opensearch-project/opensearch-build/blob/1499c472fec3d36bc0d3b30ffca8b08bb5a65c4d/RELEASE_PROCESS_OPENSEARCH.md) soon. CC: @prudhvigodithi
> 
> @bbarani should this be part of the https://github.com/opensearch-project/.github/blob/main/RELEASING.md file as the file `RELEASE_PROCESS_OPENSEARCH.md` can focus on just the release steps.

I am good with that. Lets try to combine both motivation and summary in to one section and also add roles / responsibilities to the same doc.

**Comment by prudhvigodithi** [comment]
> Release manager is responsible for creating [child component issues](https://github.com/prudhvigodithi/opensearch-build/blob/public/RELEASE_PROCESS_OPENSEARCH.md#components-release-issue) on all Github repositories participating in a release.
> * Can we automate the process if it isn't already? If it is automated, is there a doc the RM can use to show them how to do this?

This process is automated now, along with the global release issue created in the build repo (Example for [3.0.0](https://github.com/opensearch-project/opensearch-build/issues/3747)), the individual component (plugin repos) release issues are also auto-created and links back to the global release issue part of the build repo (Example for [3.0.0 component release issues](https://github.com/issues?q=is%3Aopen+is%3Aissue+user%3Aopensearch-project+%5BRELEASE%5D+Release+version+3.0.0+in%3Atitle+)).

**Comment by gaiksaya** [comment]
Hello everyone,
I created an issue that describes the current process, the gaps as well as an approach to make the release process more efficient and hands-free. 
Please take a look and provide feedback. 
Thanks!
https://github.com/opensearch-project/opensearch-build/issues/5171
